compiler/il/symbol/OMRResolvedMethodSymbol.cpp

```diff
@@ -856,6 +856,20 @@ OMR::ResolvedMethodSymbol::genAndAttachOSRCodeBlocks(int32_t currentInlinedSiteI
 
    TR::TreeTop *lastTreeTop = NULL;
    TR_OSRMethodData *osrMethodData = self()->comp()->getOSRCompilationData()->findOrCreateOSRMethodData(currentInlinedSiteIndex, self());
+
+   // Under OSR, the method entry for the outermost method is an implicit OSR point
+   // To ensure a transition from this point is possible, it is necessary to generate and attach
+   // a catch and code block now, with an exception edge to the catch block from the first block
+   //
+   bool genForOuterEntry = self()->comp()->isOutermostMethod() && self()->comp()->getHCRMode() == TR::osr;
```

Comment:
What about FSD and other OSR modes? The comment makes it sound like this should be a test for any OSR at all not just nextGenHCR